src/Raven.Pal/inc/rvn.h

```diff
@@ -60,6 +62,30 @@ struct RVN_RANGE_LIST
     size_t number_of_bytes;
 };
 
+typedef void (*MemoryLockCallback)(int64_t size, char* filename);
+typedef bool (*RecoveryMemoryLockFailureCallback)(int64_t size, char* filename);
```

Comment:
It would be good to have a clear documentation here in the typedef about what it is expected from each of the callbacks and the conditions in which they get called.